net/tstun/mtu.go

```diff
@@ -1,33 +1,168 @@
+
 package tstun
 
-import "tailscale.com/envknob"
+import (
+	"tailscale.com/envknob"
+)
+
+// The MTU (Maximum Transmission Unit) of a network interface is the
+// largest packet that can be sent or received through that interface,
+// including all headers above the link layer (e.g. IP headers, UDP
+// headers, Wireguard headers, etc.). We have to think about several
+// different values of MTU:
+//
+// Wire MTU: The MTU of an interface underneath the tailscale TUN,
+// e.g. an Ethernet network card will default to a 1500 byte MTU. The
+// user may change this MTU at any time.
+//
+// TUN MTU: The current MTU of the tailscale TUN. This MTU is adjusted
+// downward to make room for the wireguard/tailscale headers. For
+// example, if the underlying network interface's MTU is 1500 bytes,
+// the maximum size of a packet entering the tailscale TUN is 1420
+// bytes. The user may change this MTU at any time via the OS's tools
+// (ifconfig, ip, etc.).
+//
+// User configured initial MTU: The MTU the tailscale TUN should be
+// created with, set by the user via TS_DEBUG_MTU. It should be
+// adjusted down from the underlying interface MTU by 80 bytes to make
+// room for the wireguard headers. This envknob is mostly for
+// debugging. This value is used once at TUN creation and ignored
+// thereafter.
+//
+// User configured current MTU: The MTU set via the OS's tools
+// (ifconfig, ip, etc.). This MTU can change at any time. Setting the
+// MTU this way goes through the MTU() method of tailscale's TUN
+// wrapper.
+//
+// Maximum probed MTU: This is the largest MTU size that we send probe
+// packets for.
+//
+// Safe MTU: If the tailscale TUN MTU is set to this value, almost all
+// packets will get to their destination. Tailscale defaults to this
+// MTU in the absence of path MTU probe information or user MTU
+// configuration. We may occasionally find a path that needs a smaller
+// MTU but it is very rare.
+//
+// Peer MTU: This is the path MTU to a peer's current best
+// endpoint. It defaults to the Safe MTU unless we have path MTU probe
+// results that tell us otherwise.
+//
+// Initial MTU: This is the MTU tailscaled creates the TUN with. In
+// order of priority, it is:
+//
+// 1. If set, the value of TS_DEBUG_MTU clamped to a maximum of 65536
+// 2. If TS_DEBUG_ENABLE_PMTUD is set, the maximum size MTU we probe, minus wg overhead
+// 3. If TS_DEBUG_ENABLE_PMTUD is not set, the Safe MTU
+//
+// Current MTU: This the MTU of the tailscale TUN at any given moment
+// after TUN creation. In order of priority, it is:
+//
+// 1. The MTU set by the user via the OS, if it has ever been set
+// 2. If TS_DEBUG_ENABLE_PMTUD is set, the maximum size MTU we probe, minus wg overhead
+// 4. If TS_DEBUG_ENABLE_PMTUD is not set, the Safe MTU
+
+// TunMTU is used to describe MTUs for the tailscale TUN, while
+// WireMTU is used to describe MTUs for the underlying network
+// devices.
+type TunMTU uint32
+type WireMTU uint32
```

Comment:
Go documentation is only applied to the variable directly below a comment, so the docs look like this:
```
$ go doc ./net/tstun TunMTU
package tstun // import "tailscale.com/net/tstun"

type TunMTU uint32
    TunMTU is used to describe MTUs for the tailscale TUN, while WireMTU is used
    to describe MTUs for the underlying network devices.

func DefaultTunMTU() TunMTU
func WireToTunMTU(w WireMTU) TunMTU

$ go doc ./net/tstun WireMTU
package tstun // import "tailscale.com/net/tstun"

type WireMTU uint32

var MaxProbedWireMTU WireMTU
func DefaultWireMTU() WireMTU
func TunToWireMTU(t TunMTU) WireMTU
```

You may want to move the comment, so that documentation is applied correctly:
```suggestion
// TunMTU is used to describe MTUs for the tailscale TUN.
type TunMTU uint32
// WireMTU is used to describe MTUs for the underlying network
// devices.
type WireMTU uint32
```

Reply:
Thanks, fixed!